modules/core/src/alloc.cpp

```diff
@@ -100,25 +100,24 @@ static bool readMemoryAlignmentParameter()
     // TODO add checks for valgrind, ASAN if value == false
```

Comment:
> // should not call fastMalloc() internally

Main assumption is here to avoid recursive deadlock.

Reply:
This also holds for this PRs code. C++11 standard declares that reentrancy on the same thread that is in a static local initialization results in undefined behavior.

Reply:
Actually I didn't use proposed `static bool useMemalign = readMemoryAlignmentParameter();` form initially even under `#if CV_CXX11` on 3.4 branch.
OpenCV 3.4 supports C++98 but it allows to be compiled in C++11 mode (GCC 7.x+ uses C++14 mode by default).

In OpenCV 3.x there is own `cv::String` which uses `fastMalloc()` internally (OpenCV 4.x uses typedef on `std::string`).
getConfigurationParameterBool() uses `cv::String` which uses `fastAlloc()`.

Adding scope-based `fastMalloc()`/`fastFree()` here causes hang with this PR (on master):
```
    void* p = fastMalloc(1024);
    fastFree(p);
```
(triggers undefined behavior of dynamic initialization)

---

So this comment about assumption was not correct initially...
- for OpenCV 3.4 now
- there are plans to implement configuration manager (but probably such manager is not usable in this place if it will use `fastMalloc()`).